src/compas_timber/design/workflow.py

```diff
@@ -91,23 +91,24 @@ def joints_from_beams_and_rules(elements, rules, max_distance=TOL.absolute):
         element_pairs = solver.find_intersecting_pairs(elements, rtree=True, max_distance=max_rule_distance)
         joint_defs = []
         unmatched_pairs = []
+        compliant_direct_rules = []
+        for rule in direct_rules:
+            if rule.comply(rule.elements, model_max_distance=max_distance):  # see if pair complies with max distance
```

Comment:
if rule already contains the elements, why does do they need to be passed from the outside?